### DISCUSSION TITLE
read pcf8574  inputs with berry script

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#18180] [Q&A]
to check which button ist pressed, i want to use one rule like this:

``` 
tasmota.add_rule("PCF8574-3_INP#",  which_button_pressed)
```
or better:

``` 
tasmota.add_rule("PCF8574",  which_button_pressed)
```

instead of several rules like this:

```
tasmota.add_rule("PCF8574-3_INP#Dx=0",  button_x_pressed)
```

is there an option to read the pcf8574 pin states in berry script?

Or is it possible to get the last full message 

( 
``` 
02:46:53.316 RSL: RESULT = {"Time":"2023-03-13T02:46:53","PCF8574-3_INP":{"D7":0}
```
}

after triggering with

``` 
tasmota.add_rule("PCF8574",  which_button_pressed)
```
so i could parse the string to check with button is pressed or released?

i would like to use only one rule for all inputs instead off 1 rule for each.

thanks alot!

**Comment by barbudor** [comment]
With your latest rule, you would get the complete message as a map that you can parse in the rule function
https://tasmota.github.io/docs/Berry/#rules

**Reply by ChrisVonFrosta** [reply]
i know this side [https://tasmota.github.io/docs/Berry/#rules](https://tasmota.github.io/docs/Berry/#rules) but don't see how to get the full message from the message that triggered.

Where i got the complete Message after that rule
```
tasmota.add_rule("PCF8574",  which_button_pressed)
```
was triggered?

it would be so great if we could request the input stats like gpio_read()

**Reply by barbudor** [reply]
The 3rd parameter of your 'which_button_pressed' function

**Reply by ChrisVonFrosta** [reply]
my function which_button_pressed has no parameters?! i can't follow you.
or do pass the triggered rule some parameter automaticly? 
how to acces the "3rd" parameter you are talking about?

**Reply by ChrisVonFrosta** [reply]
![grafik](https://user-images.githubusercontent.com/16272493/224679273-8fd4e122-9a6e-4f9b-8d62-815f2733ed60.png)
 i see...
